common4j/src/main/com/microsoft/identity/common/java/eststelemetry/SchemaConstants.java

```diff
@@ -64,42 +77,132 @@ public static final class Value {
+     * We have lots of code that is executed outside of a DiagnosticContext i.e.
+     * ThreadLocal is not populated with a correlation Id. Since the current telemetry design is
+     * strictly around DiagnosticContext, therefore we need this supplemental cache to capture these
+     * fields that are emitted in code that is running outside that context.
+     */
+    private static final List<String> allowedFieldsForOfflineEmit = Arrays.asList(
+            Key.FLW_SIGNIN_APP,
+            Key.FLW_SIGNOUT_APP
+    );
 
+    /**
+     * Indicates if the supplied field is part of the platform field schema for current request.
+     *
+     * @param key the key that needs to be checked
+     * @return a boolean indicating if the field if part of current request platform schema
+     */
     static boolean isCurrentPlatformField(final String key) {
-        return Arrays.asList(currentRequestPlatformFields).contains(key);
+        return currentRequestAndroidPlatformFields.contains(key) ||
+                currentRequestSharedFlwPlatformFieldsForAndroidAndiOSBroker.contains(key) ||
+                currentRequestSharedMultipleWpjPlatformFieldsForAndroidAndiOSBroker.contains(key);
     }
 
+    /**
+     * Indicates if the supplied field is part of the platform field schema for last request.
+     *
+     * @param key the key that needs to be checked
+     * @return a boolean indicating if the field if part of last request platform schema
+     */
     static boolean isLastPlatformField(final String key) {
-        return Arrays.asList(lastRequestPlatformFields).contains(key);
+        return lastRequestPlatformFields.contains(key);
+    }
+
+    /**
+     * Indicates if this field is allowed to be emitted outside of a DiagnosticContext.
+     * We have lots of code that is executed outside of a DiagnosticContext i.e.
+     * ThreadLocal is not populated with a correlation Id. Since the current telemetry design is
+     * strictly around DiagnosticContext, therefore we need this supplemental cache to capture these
+     * fields that are emitted in code that is running outside that context.
+     */
+    static boolean isOfflineEmitAllowedForThisField(final String key) {
+        return allowedFieldsForOfflineEmit.contains(key);
     }
 
+    /**
+     * Get all the "ordered" platform fields for the current request.
+     *
+     * @param isSharedDeviceScenario a boolean that indicates if we're capturing telemetry in shared
+     *                               device scenario
+     * @return a {@link List} of ordered current request platform fields
+     */
+    static List<String> getCurrentRequestPlatformFields(final boolean isSharedDeviceScenario) {
+        final List<String> consolidatedPlatformFields = new ArrayList<>();
+
+        if (isSharedDeviceScenario) {
+            consolidatedPlatformFields.addAll(currentRequestSharedFlwPlatformFieldsForAndroidAndiOSBroker);
```

Comment:
As synced offline: we might want to specify platform field schema versioning (i.e. A1 for android version 1, I1 for ios version 1) - such that we know how to query data for each versions.

Alternatively, we can remove the concept of 'shared platform fields' and just keep appending new fields behind. That way, we won't break backward compatibility.

Reply:
Design issue - feedback provided in design PR here: https://dev.azure.com/IdentityDivision/DevEx/_git/AuthLibrariesApiReview/pullrequest/5168?path=/Telemetry/MSALServerSideTelemetry.md

Reply:
I am also not sure if we want to maintain common telemetry fields (iOS and Android) separately here.
instead of just capturing all the Android related fields as one list.

Reply:
@iamgusain I'm not sure I understand your comment. Can you elaborate?

Reply:
@rpdome Added platform schema version here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1799/commits/8682bc372ed773a3aec645c9714e8455a07fd908